packages/job-worker/src/playout/playout.ts

```diff
@@ -837,140 +795,140 @@ export async function onPartPlaybackStarted(context: JobContext, data: OnPartPla
 			if (!playlist.activationId) throw new Error(`Rundown Playlist "${data.playlistId}" is not active!`)
 		},
 		async (cache) => {
-			const playingPartInstance = cache.PartInstances.findOne(data.partInstanceId)
-			if (!playingPartInstance)
-				throw new Error(
-					`PartInstance "${data.partInstanceId}" in RundownPlayst "${data.playlistId}" not found!`
-				)
-
-			// make sure we don't run multiple times, even if TSR calls us multiple times
-			const isPlaying =
-				playingPartInstance.timings?.startedPlayback && !playingPartInstance.timings?.stoppedPlayback
-			if (!isPlaying) {
-				logger.debug(
-					`Playout reports PartInstance "${data.partInstanceId}" has started playback on timestamp ${new Date(
-						data.startedPlayback
-					).toISOString()}`
-				)
-
-				const playlist = cache.Playlist.doc
-
-				const rundown = cache.Rundowns.findOne(playingPartInstance.rundownId)
-				if (!rundown) throw new Error(`Rundown "${playingPartInstance.rundownId}" not found!`)
-
-				const { currentPartInstance, previousPartInstance } = getSelectedPartInstancesFromCache(cache)
-
-				if (playlist.currentPartInstanceId === data.partInstanceId) {
-					// this is the current part, it has just started playback
-					if (playlist.previousPartInstanceId) {
-						if (!previousPartInstance) {
-							// We couldn't find the previous part: this is not a critical issue, but is clearly is a symptom of a larger issue
-							logger.error(
-								`Previous PartInstance "${playlist.previousPartInstanceId}" on RundownPlaylist "${playlist._id}" could not be found.`
-							)
-						} else if (!previousPartInstance.timings?.duration) {
-							onPartHasStoppedPlaying(cache, previousPartInstance, data.startedPlayback)
-						}
-					}
+			await _onPartPlaybackStarted(context, cache, data)
+		}
+	)
+}
+async function _onPartPlaybackStarted(context: JobContext, cache: CacheForPlayout, data: OnPartPlaybackStartedProps) {
```

Comment:
This method is almost 130 lines long.
A good rule of thumb is to try and keep them around 25-30 lines.
A good place to start refactoring this to smaller methods would be to look at your comments that explain what is happening and extract those sections.

Reply:
Ive never gotten along with the number of lines rule. Ive often found it to want me to split a method with no logical split point, or resulting in a method that is then problematic because of number of parameters.
And it also makes it harder for future improvements when you to change how data flows between these methods.

I agree that some of the methods on Sofie are way too long, but Im not convinced this is one of those

Reply:
> Ive often found it to want me to split a method with no logical split point

Hence it's a rule of thumb and not a set in stone rule :) 

> makes it harder for future improvements

I would argue that having these long method makes it even harder for future improvements. A variable that is used multiple times over the span of 100+ lines becomes near impossible to keep track of and makes it so much easier to introduce unknown side effects. 
Also, the sheer amount of logic you need to consider when dealing with a long method compared to a small method makes the method more brittle, makes it more difficult for newer team members to understand the method and increases the amount of time it takes to understand the method well enough to make changes to it.

> resulting in a method that is then problematic because of number of parameters

I would could this a side effect of having the long methods. Having these longs methods where we have access to our local variables all of the time allows us to be "lazy" with our design. Since we just have everything available all of the time we start using everything all of time, making "spaghetti code" within a single function which makes it a parameter hell trying to extract the long methods into smaller methods.

> but Im not convinced this is one of those

Let's agree to disagree then :)

Reply:
If you want more opinions than mine then I would highly recommend that you read "Clean Code" by Robert C. Martin if you haven't allready.

Reply:
I'll make sure to bring this up at our upcoming talk.